src/utils/proxy.mjs

```diff
@@ -513,11 +484,46 @@ const initializeProxy = async function ({ configPath, distDir, env, host, port,
   return handlers
 }
 
+/**
+ *
+ * @param {Record<string, any>} options
+ * @param {IncomingMessage} req
+ * @param {ServerResponse} res
+ *
+ * @returns {Promise<void>}
+ */
 const onRequest = async (
   { addonsUrls, edgeFunctionsProxy, env, functionsServer, proxy, rewriter, settings, siteInfo },
   req,
   res,
 ) => {
+  res.setHeader('server', 'Netlify')
+
+  if (req.method === 'GET') {
+    const url = reqToURL(req)
+    const decodedUrl = decodeURIComponent(url.pathname)
+    const staticFile = await getStatic(decodedUrl, settings.dist)
+
+    if (staticFile) {
+      const unixStaticFile = unixify(staticFile)
+      if (decodedUrl !== unixStaticFile && (unixStaticFile.endsWith('.html') || unixStaticFile.endsWith('.htm'))) {
+        const canonicalPath = getCanonicalURLPath(unixStaticFile)
+
+        if (canonicalPath !== decodedUrl) {
+          console.log(`${NETLIFYDEVLOG} Redirecting to canonical URL ${decodedUrl} -> ${canonicalPath}`)
+          res.setHeader('age', '0')
```

Comment:
Feels a bit odd to be repeating the code where we set these headers. Do you think it would make sense to centralise that somehow?

Reply:
Problem is that this is the default Response object which uses `setHeader()`. In the static server it uses a FastifyResposne object which uses `header()`.